creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -176,13 +176,26 @@ class SnapbackSM {
         `stateMachineQueue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
       )
     })
+
+    // Re-queue state machine job when the current job fails or succeeds
```

Comment:
so one thing to note is that i believe these events are after the redis state of a job is updated
since its just an async function on the main node process and a promise cannot be canceled, the old code may still be running even if bull thinks the job is complete or stopped for any reason

basically my point here is we may be susceptible to having 2 jobs running code at the same time, even tho bull only thinks 1 is

i dont think i've seen this yet but something to watch out for

Reply:
huh interesting, haven't seen this yet but I'll keep an eye out in the logs / dashboard

Reply:
i believe this is mostly true for a stalled job bc any other outcome means the promise settled one way or another